spike_main/spike.cc

```diff
@@ -246,7 +242,8 @@ static std::vector<std::pair<reg_t, mem_t*>> make_mems(const std::vector<mem_cfg
   std::vector<std::pair<reg_t, mem_t*>> mems;
   mems.reserve(layout.size());
   for (const auto &cfg : layout) {
-    mems.push_back(std::make_pair(cfg.base, new mem_t(cfg.size)));
+    size_t size = cfg.get_inclusive_end() - cfg.get_base() + 1;
```

Comment:
`size()` should be a method on `mem_cfg_t`. I know you want to discourage use of `size` directly but there are legit uses of it, and this smells of Feature Envy.

Reply:
addressed